chat-plugins/mafia.js

```diff
@@ -396,8 +399,31 @@ class MafiaTracker extends Rooms.RoomGame {
 		return true;
 	}
 
-	getHammer() {
-		return Math.floor(Object.keys(this.players).length / 2) + 1;
+	resetHammer() {
+		this.setHammer(Math.floor(Object.keys(this.players).length / 2) + 2);
+	}
+
+	setHammer(count) {
+		this.hammerCount = count;
+		this.sendRoom(`The hammer count at ${this.hammerCount}, and lynches have been reset.`, {declare: true});
+		this.lynches = Object.create(null);
+		this.hasPlurality = null;
+		for (const player of Object.values(this.players)) {
+			player.lynching = '';
+		}
+	}
+
+	shiftHammer(count) {
+		this.hammerCount = count;
+		this.sendRoom(`The hammer count has been shifted to ${this.hammerCount}. Lynches have not been reset.`, {declare: true});
+		let hammered = [];
+		for (const lynch in this.lynches) {
+			if (this.lynches[lynch].count >= this.hammerCount) hammered.push(lynch === 'nolynch' ? 'Nobody' : lynch);
+		}
+		if (hammered.length) {
+			this.sendRoom(`${hammered.length} player${hammered.length > 1 ? "s have " : " has "}been hammered; ${hammered.join(', ')}`, {declare: true});
```

Comment:
`Chat.plural` exists

Reply:
is it worth using that when it does some checks that aren't necessary? it's basically no optimisation, but it's _free_

Reply:
Functions are always more about readability/maintainability than perf. A lot of PS would be one big function, if perf were the only priority.

Reply:
fair enough